services/webdav/pkg/service/v0/search.go

```diff
@@ -102,8 +104,39 @@ func multistatusResponse(ctx context.Context, matches []*searchmsg.Match) ([]byt
 }
 
 func matchToPropResponse(ctx context.Context, match *searchmsg.Match) (*propfind.ResponseXML, error) {
+	// unfortunately search uses own versions of ResourceId and Ref. So we need to assert them here
+	var (
+		ref string
+		err error
+	)
+
+	// to copy PROPFIND behaviour we need to deliver different ids
+	// for shares it needs to be sharestorageproviderid!shareid
+	// for other spaces it needs to be storageproviderid$spaceid
+	switch match.Entity.Ref.ResourceId.StorageId {
+	default:
+		ref, err = storagespace.FormatReference(&provider.Reference{
+			ResourceId: &provider.ResourceId{
+				StorageId: match.Entity.Ref.ResourceId.StorageId,
+				SpaceId:   match.Entity.Ref.ResourceId.SpaceId,
+			},
+			Path: match.Entity.Ref.Path,
+		})
+	case utils.ShareStorageProviderID:
+		ref, err = storagespace.FormatReference(&provider.Reference{
+			ResourceId: &provider.ResourceId{
+				//StorageId: match.Entity.Ref.ResourceId.StorageId,
+				SpaceId:  match.Entity.Ref.ResourceId.SpaceId,
+				OpaqueId: match.Entity.Ref.ResourceId.OpaqueId,
+			},
+			Path: match.Entity.Ref.Path,
+		})
+	}
+	if err != nil {
+		return nil, err
+	}
 	response := propfind.ResponseXML{
-		Href:     net.EncodePath(path.Join("/dav/spaces/", match.Entity.Ref.ResourceId.StorageId+"!"+match.Entity.Ref.ResourceId.OpaqueId, match.Entity.Ref.Path)),
+		Href:     net.EncodePath(path.Join("/remote.php/dav/spaces/", ref)),
```

Comment:
why do we have to prefix with the legacy `remote.php`?

Reply:
To be consistent with ocdav PROPFIND endpoint